Exoft.Gamification.Api.Data/Repositories/Repository.cs

```diff
@@ -66,6 +55,25 @@ public virtual void Delete(T entity)
             DbSet.Remove(entity);
         }
 
+        public async Task<ReturnPagingInfo<T>> GetPagingDataAsync(PagingInfo pagingInfo)
+        {
+            var items = await IncludeAll()
+                .Skip((pagingInfo.CurrentPage - 1) * pagingInfo.PageSize)
+                .Take(pagingInfo.PageSize)
+                .ToListAsync();
+
+            var result = new ReturnPagingInfo<T>()
+            {
+                CurrentPage = pagingInfo.CurrentPage,
+                PageSize = items.Count,
+                TotalItems = IncludeAll().Count(),
```

Comment:
Please use Count async methods.... Well prefer async methods everywhere it is possible.